src/windows/symbol.rs

```diff
@@ -374,12 +391,58 @@ impl RvaSymbols {
         frame_table: FrameTable,
     ) -> PDBSymbols {
         let mut syms = PDBSymbols::default();
+        if self.map.is_empty() {
+            return syms;
+        }
+
+        let mut ranges: BTreeMap<(u32, u32), usize> = BTreeMap::default();
+        let mut all_syms = Vec::with_capacity(self.map.len());
+
+        // First, we get all the ranges after splitting them
         for (rva, sym) in self.map.drain() {
-            syms.insert(
-                rva,
-                sym.mv_to_pdb_symbol(&dumper, address_map, &frame_table),
-            );
+            let (sym, offset) = sym.mv_to_pdb_symbol(&dumper, rva, address_map, &frame_table);
+            let last = all_syms.len();
+            if sym.len == 0 {
+                ranges.insert((rva, 0), last);
+            } else {
+                let start = offset.to_internal_rva(&address_map).unwrap();
+                let end = PdbInternalRva(start.0 + sym.len);
+                for (rva, len) in address_map
+                    .rva_ranges(start..end)
+                    .map(|r| (r.start.0, r.end.0 - r.start.0))
+                {
+                    ranges.insert((rva, len), last);
+                }
+            }
+            all_syms.push(sym);
         }
+
+        // Second, we initialize for first symbol
+        let mut iterator = ranges.iter();
+        let ((rva, len), sym_pos) = iterator.next().unwrap();
+
+        let mut last_rva = *rva;
+        let mut last_len = *len;
+        let mut last_sym = &all_syms[*sym_pos];
+        let mut last_id = last_sym.id;
+
+        // Third, we merge ranges ([a; b] + [c; d] = [a; d]) which consecutively have the same function id
+        // So the hole between [a; b] and [c; d] will become a part of the range for the function
```

Comment:
nit: Likewise. Note that by this point you have three arguments, eight local variables and two loop variables. That's a lot of stuff in a single function body.